Shared/Samples/Show device location with NMEA data sources/ShowDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,296 @@
+        )
+        
+        /// A format style for concatenating an array of text.
+        private let listFormatStyle = ListFormatStyle<StringStyle, Array>.list(type: .and, width: .narrow)
+        
+        /// The protocols used in this sample to get NMEA sentences.
+        /// They are also specified in the `Info.plist` to allow the app to
+        /// communicate with external accessory hardware.
+        private let supportedProtocolStrings: Set = [
+            "com.bad-elf.gps",
+            "com.eos-gnss.positioningsource",
+            "com.geneq.sxbluegpssource"
+        ]
+        
+        /// The tasks created during starting the data source.
+        private var tasks = [Task<Void, Never>]()
+        
+        // MARK: Methods
+        
+        deinit {
+            clearTasks()
+        }
+        
+        /// Cancels and removes tasks.
+        private func clearTasks() {
+            tasks.forEach { task in
+                task.cancel()
+            }
+            tasks.removeAll()
+        }
+        
+        /// Gets the first connected and supported Bluetooth accessory with its
+        /// protocol string.
+        /// - Returns: A tuple of the accessory and its protocol,
+        /// or `nil` if no supported accessory exists.
+        func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+            for accessory in EAAccessoryManager.shared().connectedAccessories {
+                // The protocol string to establish the EASession.
+                guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                    // Skip the accessories with protocol not for NMEA data transfer.
+                    continue
+                }
+                // Only return the first connected and supported accessory.
+                return (accessory, protocolString)
+            }
+            return nil
+        }
+        
+        /// The Bluetooth accessory picker connected to a supported accessory.
+        func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+            if let dataSource = NMEALocationDataSource(accessory: connectedAccessory, protocol: protocolString) {
+                nmeaLocationDataSource = dataSource
+                start()
+            }
+        }
+        
+        /// Resets the sample, stops the data source, cancels tasks, and resets button states and status strings.
+        func reset() {
+            // Reset buttons states.
+            isResetButtonDisabled = true
+            isSourceMenuDisabled = false
+            
+            // Reset the status text.
+            accuracyStatus = "Accuracy info will be shown here."
+            satelliteStatus = "Satellites info will be shown here."
+            
+            // Reset NMEA location data source.
+            nmeaLocationDataSource = nil
+            autoPanMode = .off
+            
+            Task {
```

Comment:
Do we usually want to use a detached task by default?

Reply:
Good catch, no.